### ISSUE TITLE
Regression? tsc=reliable not found in cmdline file - causes bad performance of PHP

### ISSUE THREAD

**Issue by kerryj89** [docker/for-mac#6430]
- [ x ] I have tried with the latest version of Docker Desktop
  - [ ] I have tried disabling enabled experimental features
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID:

### Expected behavior
I expect PHP to run smoothly after sleep.  Fix (https://github.com/docker/for-mac/issues/3455#issuecomment-1092529479) doesn't seem to be included in `/Applications/Docker.app/Contents/Resources/linuxkit/cmdline`. Consider making this an option if adding this argument is not a safe default.

### Actual behavior
PHP runs slow after waking up from sleep. Restarting Docker fixes it.

### Information

  - macOS Version: 12.4
  - Intel chip or Apple chip: Intel
  - Docker Desktop Version: 4.10.1

### Output of `/Applications/Docker.app/Contents/MacOS/com.docker.diagnose check`
```
[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0031: does the Docker API work?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0001: is the application running?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0017: can a VM be started?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0003: is the Docker CLI working?
[PASS] DD0013: is the $PATH ok?
[PASS] DD0007: is the backend responding?
[PASS] DD0014: are the backend processes running?
[PASS] DD0008: is the native API responding?
[PASS] DD0009: is the vpnkit API responding?
[PASS] DD0010: is the Docker API proxy responding?
[FAIL] DD0012: is the VM networking working? network checks failed: failed to ping host: exit status 1
[2022-07-28T13:36:29.781628000Z][com.docker.diagnose][I] ipc.NewClient: 944804d9-diagnose-network -> diagnosticd.sock diagnosticsd
[common/pkg/diagkit/gather/diagnose.runIsVMNetworkingOK()
[	common/pkg/diagkit/gather/diagnose/network.go:34 +0xd9
[common/pkg/diagkit/gather/diagnose.(*test).GetResult(0x4d442a0)
[	common/pkg/diagkit/gather/diagnose/test.go:46 +0x43
[common/pkg/diagkit/gather/diagnose.Run.func1(0x4d442a0)
[	common/pkg/diagkit/gather/diagnose/run.go:17 +0x5a
[common/pkg/diagkit/gather/diagnose.walkOnce.func1(0x2?, 0x4d442a0)
[	common/pkg/diagkit/gather/diagnose/run.go:140 +0x77
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x1, 0x4d442a0, 0xc0005a7728)
[	common/pkg/diagkit/gather/diagnose/run.go:146 +0x36
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x0, 0x4?, 0xc0005a7728)
[	common/pkg/diagkit/gather/diagnose/run.go:149 +0x73
[common/pkg/diagkit/gather/diagnose.walkOnce(0x46f5c60?, 0xc00061f888)
[	common/pkg/diagkit/gather/diagnose/run.go:135 +0xcc
[common/pkg/diagkit/gather/diagnose.Run(0x4d44420, 0x10?, {0xc00061fb18, 0x1, 0x1})
[	common/pkg/diagkit/gather/diagnose/run.go:16 +0x1cb
[main.checkCmd({0xc0001ae010?, 0x6?, 0x4?}, {0x0, 0x0})
[	common/cmd/com.docker.diagnose/main.go:133 +0x105
[main.main()
[	common/cmd/com.docker.diagnose/main.go:99 +0x2a8
[2022-07-28T13:36:29.786886000Z][com.docker.diagnose][I] (d1dcf86a) 944804d9-diagnose-network C->S diagnosticsd POST /check-network-connectivity: {"ips":["10.10.60.42","10.10.80.210"]}
[2022-07-28T13:36:30.309425000Z][com.docker.diagnose][W] (d1dcf86a) 944804d9-diagnose-network C<-S 799a365f-diagnosticsd POST /check-network-connectivity (522.606354ms): failed to ping host: exit status 1

[PASS] DD0032: do Docker networks overlap with host IPs?
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0019: is the com.docker.vmnetd process responding?
[PASS] DD0033: does the host have Internet access?

Please investigate the following 1 issue:

1 : The test: is the VM networking working?
    Failed with: network checks failed: failed to ping host: exit status 1

VM seems to have a network connectivity issue. Please check your host firewall and anti-virus settings in case they are blocking the VM.
```

### Steps to reproduce the behavior

  1. Put computer to sleep for at least several hours
  2. Wake up computer and notice PHP being slow

**Comment by nicks** [comment]
@kerryj89 Can you upload the diagnostics and post your diagnostics ID? We need the diagnostics id to be able to investigate the problem.

Re: "Fix (https://github.com/docker/for-mac/issues/3455#issuecomment-1092529479) doesn't seem to be included in /Applications/Docker.app/Contents/Resources/linuxkit/cmdline"

That doesn't mean much. I would not expect tsc=reliable to show up in that file. That's the base cmdline; the final cmd is constructed from a few sources. The proper way to check the linuxkit flags is:

```
ps aux | grep linuxkit
```

and see if tsc=reliable shows up there.

**Comment by kerryj89** [comment]
@nicks Diagnostic id: `17AEC960-790B-43DB-8A80-03C2E43053A9/20220729183759`

With `tsc=reliable` in `/Applications/Docker.app/Contents/Resources/linuxkit/cmdline` (tsc shows up):
```
kjohnson@Kerrys-MBP ~ % ps aux | grep linuxkit                                                                  
kjohnson         51623   0.0  0.0 34122844    860 s000  S+    1:22PM   0:00.00 grep linuxkit
kjohnson         46662   0.0  0.1 34967924  17212   ??  S     1:20PM   0:00.87 /Applications/Docker.app/Contents/MacOS/com.docker.virtualization --kernel /Applications/Docker.app/Contents/Resources/linuxkit/kernel --cmdline page_poison=1 vsyscall=emulate panic=1 nospec_store_bypass_disable noibrs noibpb no_stf_barrier mitigations=off tsc=reliable linuxkit.unified_cgroup_hierarchy=1 console=hvc0  irqaffinity=0 vpnkit.connect=tcp+bootstrap+client://192.168.65.2:63278/412514029e65ccbe83d44281d04099bbb1030051c991dc6731f92a618182c29e vpnkit.disable=osxfs-data --ramdisk /Applications/Docker.app/Contents/Resources/linuxkit/initrd.img --disk /Users/kjohnson/Library/Containers/com.docker.docker/Data/vms/0/data/Docker.raw --macaddr-filename vms/0/macaddr --cpus 4 --memoryMiB 8192 --console-log log/vm/console.log --console vms/0/console.sock --watchdog --vpnkit vpnkit.eth.sock --virtiofs /Users --virtiofs /Volumes --virtiofs /private --virtiofs /tmp --virtiofs /var/folders
```

Without (tsc doesn't show up):
```
kjohnson@Kerrys-MBP icpv2 % ps aux | grep linuxkit
kjohnson         45776   0.0  0.0 34132060    904 s008  S+    1:17PM   0:00.00 grep linuxkit
kjohnson         37399   0.0  0.1 34967400  15496   ??  S     1:05PM   0:01.89 /Applications/Docker.app/Contents/MacOS/com.docker.virtualization --kernel /Applications/Docker.app/Contents/Resources/linuxkit/kernel --cmdline page_poison=1 vsyscall=emulate panic=1 nospec_store_bypass_disable noibrs noibpb no_stf_barrier mitigations=off linuxkit.unified_cgroup_hierarchy=1 console=hvc0  irqaffinity=0 vpnkit.connect=tcp+bootstrap+client://192.168.65.2:62775/8aa3e6e52d38d7d06f2b1d92450211d6d249ed050f7e8002ab130eb148020ae2 vpnkit.disable=osxfs-data --ramdisk /Applications/Docker.app/Contents/Resources/linuxkit/initrd.img --disk /Users/kjohnson/Library/Containers/com.docker.docker/Data/vms/0/data/Docker.raw --macaddr-filename vms/0/macaddr --cpus 4 --memoryMiB 8192 --console-log log/vm/console.log --console vms/0/console.sock --watchdog --vpnkit vpnkit.eth.sock --virtiofs /Users --virtiofs /Volumes --virtiofs /private --virtiofs /tmp --virtiofs /var/folders
```

**Comment by nicks** [comment]
@kerryj89 can you try going to Settings -> Experimental Features, unchecking "Use the new virtualization framework", and see if that fixes the issue for you?

**Comment by kerryj89** [comment]
@nicks It won't let me uncheck that unless `Enable VirtioFS accelerated directory sharing` is disabled, and I need that to keep Docker speedy on my Mac.

![2022-07-29_15-30-44](https://user-images.githubusercontent.com/1977804/181838768-bd7f290a-023c-424d-8675-3893c37fd080.png)

**Comment by nicks** [comment]
👍 

for future me, here's a simpler verification case:

```
nick@sneezy scratch % docker run -d ubuntu tail -f /dev/null
cf6496adf56f4aa7a81609efef276a6c4b179bf2b54231cbda004b5c6ffd481a
nick@sneezy scratch % docker exec -ti cf6496adf56f4aa7a81609efef276a6c4b179bf2b54231cbda004b5c6ffd481a cat /sys/devices/system/clocksource/clocksource0/current_clocksource
tsc
[put computer into sleep]
nick@sneezy scratch % docker exec -ti cf6496adf56f4aa7a81609efef276a6c4b179bf2b54231cbda004b5c6ffd481a cat /sys/devices/system/clocksource/clocksource0/current_clocksource
hpet
```

**Comment by nicks** [comment]
this should be fixed in v4.12
